src/feasibility-form-nls.jl

```diff
@@ -39,38 +39,41 @@ end
 Converts a nonlinear least-squares problem with residual `F(x)` to a nonlinear
 optimization problem with constraints `F(x) = r` and objective `¹/₂‖r‖²`.
 """
-function FeasibilityFormNLS(nls::AbstractNLSModel; name = "$(nls.meta.name)-ffnls")
+function FeasibilityFormNLS(
+  nls::AbstractNLSModel{T, S};
+  name = "$(nls.meta.name)-ffnls",
+) where {T,S}
   nequ = nls.nls_meta.nequ
   meta = nls.meta
   nvar = meta.nvar + nequ
   ncon = meta.ncon + nequ
   nnzh = nls.nls_meta.nnzh + nequ + (meta.ncon == 0 ? 0 : meta.nnzh) # Some indexes can be repeated
-  meta = NLPModelMeta(
+  meta = NLPModelMeta{T, S}(
     nvar,
-    x0 = [meta.x0; zeros(nequ)],
-    lvar = [meta.lvar; fill(-Inf, nequ)],
-    uvar = [meta.uvar; fill(Inf, nequ)],
+    x0 = [meta.x0; zeros(T, nequ)],
+    lvar = [meta.lvar; fill(T(-Inf), nequ)],
```

Comment:
Do these have to be generic too?

Reply:
They do, because of the type constraint on `x0`, `lvar` and `uvar`.

Reply:
Doesn't line 54 create a Vector though?

Reply:
I see, so in these case, you suggest allocating the vector first. I will update.